generate persistence.xml from template with custom ant parser runner that saves generated persistence.xml to output TCK source tree

See the [first comment for an example of the error](https://github.com/eclipse-ee4j/jakartaee-tck-tools/issues/100#issuecomment-2286847017) that we see when refactoring the jpa tests that depend on the [jakartaeetck/src/com/sun/ts/tests/jpa/common/persistence-import.xml](https://github.com/jakartaee/platform-tck/blob/10.0.x/src/com/sun/ts/tests/jpa/common/persistence-import.xml) to be transformed into a persistence.xml to be included in the test that looks like the [second comment](https://github.com/eclipse-ee4j/jakartaee-tck-tools/issues/100#issuecomment-2286856105).

The same error happens for TCK classes:
- ee.jakarta.tck.persistence.core.StoredProcedureQuery.Client1
- ee.jakarta.tck.persistence.core.StoredProcedureQuery.Client2 ([link](https://github.com/eclipse-ee4j/jakartaee-tck-tools/issues/100#issuecomment-2286847017))
- ee.jakarta.tck.persistence.core.annotations.elementcollection.Client1 ([link](https://github.com/eclipse-ee4j/jakartaee-tck-tools/issues/100#issuecomment-2286862499))
- ee.jakarta.tck.persistence.core.annotations.elementcollection.Client2
- ee.jakarta.tck.persistence.core.entitytest.apitests.Client ([link](https://github.com/eclipse-ee4j/jakartaee-tck-tools/issues/100#issuecomment-2286866795))

```
INFO: Parsing(/home/smarlow/tck/tck10/jakartaeetck/src/com/sun/ts/tests/jpa/core/annotations/collectiontable/build.xml)

Aug 13, 2024 1:49:34 PM tck.jakarta.platform.ant.PackageTarget execute
INFO: Executing package target dependencies: []
Aug 13, 2024 1:49:34 PM tck.jakarta.platform.rewrite.GenerateNewTestClassRecipe$testClassVisitor visitClassDeclaration
INFO: TODO: due to The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.vehicles.xml:239: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.vehicles.xml:1266: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.ear.xml:289: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.common.xml:789: Problem reading /home/smarlow/tck/tck10/jakartaeetck/dist/com/sun/ts/tests/jpa/core/annotations/collectiontable/jpa_core_annotations_collectiontable_vehicles.ear class ee.jakarta.tck.persistence.core.annotations.collectiontable.Client couldn't be processed.
/home/smarlow/tck/tck10/jakartaeetck/src/com/sun/ts/tests/jpa/core/annotations/collectiontable/build.xml:42: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.vehicles.xml:239: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.vehicles.xml:1266: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.ear.xml:289: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.common.xml:789: Problem reading /home/smarlow/tck/tck10/jakartaeetck/dist/com/sun/ts/tests/jpa/core/annotations/collectiontable/jpa_core_annotations_collectiontable_vehicles.ear
	at org.apache.tools.ant.ProjectHelper.addLocationToBuildException(ProjectHelper.java:582)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:394)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.Target.execute(Target.java:449)
	at org.apache.tools.ant.Target.performTasks(Target.java:470)
	at tck.jakarta.platform.ant.PackageTarget.execute(PackageTarget.java:552)
	at tck.jakarta.platform.ant.api.TestPackageInfoBuilder.parseVehiclePackage(TestPackageInfoBuilder.java:321)
	at tck.jakarta.platform.ant.api.TestPackageInfoBuilder.buildTestClientsEx(TestPackageInfoBuilder.java:213)
	at tck.jakarta.platform.ant.api.TestPackageInfoBuilder.buildTestPackgeInfoEx(TestPackageInfoBuilder.java:116)
	at tck.jakarta.platform.rewrite.GenerateNewTestClassRecipe$testClassVisitor.visitClassDeclaration(GenerateNewTestClassRecipe.java:188)
	at tck.jakarta.platform.rewrite.GenerateNewTestClassRecipe$testClassVisitor.visitClassDeclaration(GenerateNewTestClassRecipe.java:102)
	at org.openrewrite.java.tree.J$ClassDeclaration.acceptJava(J.java:1286)
	at org.openrewrite.java.tree.J.accept(J.java:59)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:283)
	at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:366)
	at org.openrewrite.java.JavaVisitor.lambda$visitCompilationUnit$10(JavaVisitor.java:501)
	at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
	at org.openrewrite.java.JavaVisitor.visitCompilationUnit(JavaVisitor.java:501)
	at org.openrewrite.java.JavaIsoVisitor.visitCompilationUnit(JavaIsoVisitor.java:113)
	at org.openrewrite.java.JavaIsoVisitor.visitCompilationUnit(JavaIsoVisitor.java:30)
	at org.openrewrite.java.tree.J$CompilationUnit.acceptJava(J.java:1560)
	at org.openrewrite.java.tree.J.accept(J.java:59)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:283)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:184)
	at org.openrewrite.scheduling.RecipeRunCycle.lambda$editSources$5(RecipeRunCycle.java:164)
	at io.micrometer.core.instrument.AbstractTimer.recordCallable(AbstractTimer.java:175)
	at org.openrewrite.table.RecipeRunStats.recordEdit(RecipeRunStats.java:68)
	at org.openrewrite.scheduling.RecipeRunCycle.lambda$editSources$6(RecipeRunCycle.java:161)
	at org.openrewrite.scheduling.RecipeStack.reduce(RecipeStack.java:57)
	at org.openrewrite.scheduling.RecipeRunCycle.lambda$editSources$7(RecipeRunCycle.java:134)
	at org.openrewrite.internal.InMemoryLargeSourceSet.lambda$edit$0(InMemoryLargeSourceSet.java:66)
	at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
	at org.openrewrite.internal.InMemoryLargeSourceSet.edit(InMemoryLargeSourceSet.java:65)
	at org.openrewrite.scheduling.RecipeRunCycle.editSources(RecipeRunCycle.java:133)
	at org.openrewrite.RecipeScheduler.runRecipeCycles(RecipeScheduler.java:86)
	at org.openrewrite.RecipeScheduler.scheduleRun(RecipeScheduler.java:41)
	at org.openrewrite.Recipe.run(Recipe.java:340)
	at org.openrewrite.Recipe.run(Recipe.java:336)
	at org.openrewrite.Recipe.run(Recipe.java:332)
	at org.openrewrite.maven.AbstractRewriteBaseRunMojo.runRecipe(AbstractRewriteBaseRunMojo.java:142)
	at org.openrewrite.maven.AbstractRewriteBaseRunMojo.listResults(AbstractRewriteBaseRunMojo.java:119)
	at org.openrewrite.maven.AbstractRewriteRunMojo.execute(AbstractRewriteRunMojo.java:63)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:328)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:316)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:174)
	at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:75)
	at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:162)
	at org.apache.maven.buildcache.BuildCacheMojosExecutionStrategy.execute(BuildCacheMojosExecutionStrategy.java:132)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:159)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:906)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:283)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:206)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:283)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:226)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:407)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:348)
Caused by: /home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.vehicles.xml:239: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.vehicles.xml:1266: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.ear.xml:289: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.common.xml:789: Problem reading /home/smarlow/tck/tck10/jakartaeetck/dist/com/sun/ts/tests/jpa/core/annotations/collectiontable/jpa_core_annotations_collectiontable_vehicles.ear
	at org.apache.tools.ant.ProjectHelper.addLocationToBuildException(ProjectHelper.java:582)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:394)
	at net.sf.antcontrib.logic.ForTask.doSequentialIteration(ForTask.java:259)
	at net.sf.antcontrib.logic.ForTask.doToken(ForTask.java:268)
	at net.sf.antcontrib.logic.ForTask.doTheTasks(ForTask.java:299)
	at net.sf.antcontrib.logic.ForTask.execute(ForTask.java:244)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:197)
	at jdk.internal.reflect.GeneratedMethodAccessor799.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:391)
	... 72 more
Caused by: /home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.vehicles.xml:1266: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.ear.xml:289: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.common.xml:789: Problem reading /home/smarlow/tck/tck10/jakartaeetck/dist/com/sun/ts/tests/jpa/core/annotations/collectiontable/jpa_core_annotations_collectiontable_vehicles.ear
	at org.apache.tools.ant.ProjectHelper.addLocationToBuildException(ProjectHelper.java:582)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:394)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:197)
	at jdk.internal.reflect.GeneratedMethodAccessor799.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:197)
	at jdk.internal.reflect.GeneratedMethodAccessor799.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:391)
	... 105 more
Caused by: /home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.ear.xml:289: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.common.xml:789: Problem reading /home/smarlow/tck/tck10/jakartaeetck/dist/com/sun/ts/tests/jpa/core/annotations/collectiontable/jpa_core_annotations_collectiontable_vehicles.ear
	at org.apache.tools.ant.ProjectHelper.addLocationToBuildException(ProjectHelper.java:582)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:394)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:217)
	at jdk.internal.reflect.GeneratedMethodAccessor799.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:217)
	at jdk.internal.reflect.GeneratedMethodAccessor799.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:197)
	at jdk.internal.reflect.GeneratedMethodAccessor799.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:391)
	... 148 more
Caused by: /home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.common.xml:789: Problem reading /home/smarlow/tck/tck10/jakartaeetck/dist/com/sun/ts/tests/jpa/core/annotations/collectiontable/jpa_core_annotations_collectiontable_vehicles.ear
	at org.apache.tools.ant.types.ZipScanner.fillMapsFromArchive(ZipScanner.java:81)
	at org.apache.tools.ant.types.ArchiveScanner.scanme(ArchiveScanner.java:348)
	at org.apache.tools.ant.types.ArchiveScanner.getResource(ArchiveScanner.java:285)
	at org.apache.tools.ant.util.ResourceUtils.selectSources(ResourceUtils.java:203)
	at org.apache.tools.ant.util.ResourceUtils.selectOutOfDateSources(ResourceUtils.java:149)
	at org.apache.tools.ant.util.ResourceUtils.selectOutOfDateSources(ResourceUtils.java:127)
	at org.apache.tools.ant.taskdefs.Zip.selectOutOfDateResources(Zip.java:1550)
	at org.apache.tools.ant.taskdefs.Zip.getResourcesToAdd(Zip.java:1440)
	at org.apache.tools.ant.taskdefs.Zip.getResourcesToAdd(Zip.java:1263)
	at org.apache.tools.ant.taskdefs.Jar.getResourcesToAdd(Jar.java:867)
	at org.apache.tools.ant.taskdefs.Zip.executeMain(Zip.java:671)
	at org.apache.tools.ant.taskdefs.Zip.execute(Zip.java:627)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:217)
	at jdk.internal.reflect.GeneratedMethodAccessor799.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:197)
	at jdk.internal.reflect.GeneratedMethodAccessor799.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:391)
	... 205 more
Caused by: java.util.zip.ZipException: archive is not a ZIP archive
	at org.apache.tools.zip.ZipFile.positionAtEndOfCentralDirectoryRecord(ZipFile.java:789)
	at org.apache.tools.zip.ZipFile.positionAtCentralDirectory(ZipFile.java:725)
	at org.apache.tools.zip.ZipFile.populateFromCentralDirectory(ZipFile.java:461)
	at org.apache.tools.zip.ZipFile.<init>(ZipFile.java:217)
	at org.apache.tools.zip.ZipFile.<init>(ZipFile.java:192)
	at org.apache.tools.ant.types.ZipScanner.fillMapsFromArchive(ZipScanner.java:63)
	... 259 more
```

```
Aug 13, 2024 2:02:54 PM tck.jakarta.platform.ant.PackageTarget execute
INFO: Executing package target dependencies: []
Aug 13, 2024 2:02:54 PM tck.jakarta.platform.rewrite.GenerateNewTestClassRecipe$testClassVisitor visitClassDeclaration
INFO: TODO: due to The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:107: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:291: org.jdom.input.JDOMParseException: Error on line 1: The markup in the document following the root element must be well-formed. class ee.jakarta.tck.persistence.core.StoredProcedureQuery.Client2 couldn't be processed.
/home/smarlow/tck/tck10/jakartaeetck/src/com/sun/ts/tests/jpa/core/StoredProcedureQuery/build.xml:39: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:107: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:291: org.jdom.input.JDOMParseException: Error on line 1: The markup in the document following the root element must be well-formed.
	at org.apache.tools.ant.ProjectHelper.addLocationToBuildException(ProjectHelper.java:582)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:394)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.Target.execute(Target.java:449)
	at org.apache.tools.ant.Target.performTasks(Target.java:470)
	at tck.jakarta.platform.ant.PackageTarget.execute(PackageTarget.java:552)
	at tck.jakarta.platform.ant.api.TestPackageInfoBuilder.parseVehiclePackage(TestPackageInfoBuilder.java:308)
	at tck.jakarta.platform.ant.api.TestPackageInfoBuilder.buildTestClientsEx(TestPackageInfoBuilder.java:211)
	at tck.jakarta.platform.ant.api.TestPackageInfoBuilder.buildTestPackgeInfoEx(TestPackageInfoBuilder.java:115)
	at tck.jakarta.platform.rewrite.GenerateNewTestClassRecipe$testClassVisitor.visitClassDeclaration(GenerateNewTestClassRecipe.java:183)
	at tck.jakarta.platform.rewrite.GenerateNewTestClassRecipe$testClassVisitor.visitClassDeclaration(GenerateNewTestClassRecipe.java:101)
	at org.openrewrite.java.tree.J$ClassDeclaration.acceptJava(J.java:1286)
	at org.openrewrite.java.tree.J.accept(J.java:59)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:283)
	at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:366)
	at org.openrewrite.java.JavaVisitor.lambda$visitCompilationUnit$10(JavaVisitor.java:501)
	at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
	at org.openrewrite.java.JavaVisitor.visitCompilationUnit(JavaVisitor.java:501)
	at org.openrewrite.java.JavaIsoVisitor.visitCompilationUnit(JavaIsoVisitor.java:113)
	at org.openrewrite.java.JavaIsoVisitor.visitCompilationUnit(JavaIsoVisitor.java:30)
	at org.openrewrite.java.tree.J$CompilationUnit.acceptJava(J.java:1560)
	at org.openrewrite.java.tree.J.accept(J.java:59)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:283)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:184)
	at org.openrewrite.scheduling.RecipeRunCycle.lambda$editSources$5(RecipeRunCycle.java:164)
	at io.micrometer.core.instrument.AbstractTimer.recordCallable(AbstractTimer.java:175)
	at org.openrewrite.table.RecipeRunStats.recordEdit(RecipeRunStats.java:68)
	at org.openrewrite.scheduling.RecipeRunCycle.lambda$editSources$6(RecipeRunCycle.java:161)
	at org.openrewrite.scheduling.RecipeStack.reduce(RecipeStack.java:57)
	at org.openrewrite.scheduling.RecipeRunCycle.lambda$editSources$7(RecipeRunCycle.java:134)
	at org.openrewrite.internal.InMemoryLargeSourceSet.lambda$edit$0(InMemoryLargeSourceSet.java:66)
	at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
	at org.openrewrite.internal.InMemoryLargeSourceSet.edit(InMemoryLargeSourceSet.java:65)
	at org.openrewrite.scheduling.RecipeRunCycle.editSources(RecipeRunCycle.java:133)
	at org.openrewrite.RecipeScheduler.runRecipeCycles(RecipeScheduler.java:86)
	at org.openrewrite.RecipeScheduler.scheduleRun(RecipeScheduler.java:41)
	at org.openrewrite.Recipe.run(Recipe.java:340)
	at org.openrewrite.Recipe.run(Recipe.java:336)
	at org.openrewrite.Recipe.run(Recipe.java:332)
	at org.openrewrite.maven.AbstractRewriteBaseRunMojo.runRecipe(AbstractRewriteBaseRunMojo.java:142)
	at org.openrewrite.maven.AbstractRewriteBaseRunMojo.listResults(AbstractRewriteBaseRunMojo.java:119)
	at org.openrewrite.maven.AbstractRewriteRunMojo.execute(AbstractRewriteRunMojo.java:63)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:328)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:316)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:174)
	at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:75)
	at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:162)
	at org.apache.maven.buildcache.BuildCacheMojosExecutionStrategy.execute(BuildCacheMojosExecutionStrategy.java:132)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:159)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:906)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:283)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:206)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:283)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:226)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:407)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:348)
Caused by: /home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:107: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:291: org.jdom.input.JDOMParseException: Error on line 1: The markup in the document following the root element must be well-formed.
	at org.apache.tools.ant.ProjectHelper.addLocationToBuildException(ProjectHelper.java:582)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:394)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:197)
	at jdk.internal.reflect.GeneratedMethodAccessor166.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:217)
	at jdk.internal.reflect.GeneratedMethodAccessor166.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:197)
	at jdk.internal.reflect.GeneratedMethodAccessor166.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:391)
	... 72 more
Caused by: /home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:291: org.jdom.input.JDOMParseException: Error on line 1: The markup in the document following the root element must be well-formed.
	at com.sun.ant.taskdefs.xml.XMLFragment.init(XMLFragment.java:134)
	at com.sun.ant.taskdefs.xml.ModifyXML.addConfiguredXmlfragment(ModifyXML.java:220)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.IntrospectionHelper$AddNestedCreator.istore(IntrospectionHelper.java:1519)
	at org.apache.tools.ant.IntrospectionHelper$AddNestedCreator.store(IntrospectionHelper.java:1513)
	at org.apache.tools.ant.IntrospectionHelper$Creator.store(IntrospectionHelper.java:1418)
	at org.apache.tools.ant.UnknownElement.handleChild(UnknownElement.java:601)
	at org.apache.tools.ant.UnknownElement.handleChildren(UnknownElement.java:367)
	at org.apache.tools.ant.UnknownElement.configure(UnknownElement.java:207)
	at org.apache.tools.ant.UnknownElement.maybeConfigure(UnknownElement.java:166)
	at org.apache.tools.ant.Task.perform(Task.java:349)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:197)
	at jdk.internal.reflect.GeneratedMethodAccessor166.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:391)
	... 129 more
Caused by: org.jdom.input.JDOMParseException: Error on line 1: The markup in the document following the root element must be well-formed.
	at org.jdom.input.SAXBuilder.build(SAXBuilder.java:533)
	at org.jdom.input.SAXBuilder.build(SAXBuilder.java:946)
	at com.sun.ant.taskdefs.xml.XMLFragment.init(XMLFragment.java:130)
	... 165 more
Caused by: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 49; The markup in the document following the root element must be well-formed.
	at java.xml/com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:204)
	at java.xml/com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.fatalError(ErrorHandlerWrapper.java:178)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:400)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:327)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLScanner.reportFatalError(XMLScanner.java:1465)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$TrailingMiscDriver.next(XMLDocumentScannerImpl.java:1394)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:112)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:542)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:889)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:825)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1224)
	at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:637)
	at org.jdom.input.SAXBuilder.build(SAXBuilder.java:518)
	... 167 more
```

/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:291: org.jdom.input.JDOMParseException: Error on line 1: The markup in the document following the root element must be well-formed. class ee.jakarta.tck.persistence.core.annotations.elementcollection.Client1 couldn't be processed.
/home/smarlow/tck/tck10/jakartaeetck/src/com/sun/ts/tests/jpa/core/annotations/elementcollection/build.xml:39: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:107: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:291: org.jdom.input.JDOMParseException: Error on line 1: The markup in the document following the root element must be well-formed.
	at org.apache.tools.ant.ProjectHelper.addLocationToBuildException(ProjectHelper.java:582)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:394)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.Target.execute(Target.java:449)
	at org.apache.tools.ant.Target.performTasks(Target.java:470)
	at tck.jakarta.platform.ant.PackageTarget.execute(PackageTarget.java:552)
	at tck.jakarta.platform.ant.api.TestPackageInfoBuilder.parseVehiclePackage(TestPackageInfoBuilder.java:308)
	at tck.jakarta.platform.ant.api.TestPackageInfoBuilder.buildTestClientsEx(TestPackageInfoBuilder.java:211)
	at tck.jakarta.platform.ant.api.TestPackageInfoBuilder.buildTestPackgeInfoEx(TestPackageInfoBuilder.java:115)
	at tck.jakarta.platform.rewrite.GenerateNewTestClassRecipe$testClassVisitor.visitClassDeclaration(GenerateNewTestClassRecipe.java:183)
	at tck.jakarta.platform.rewrite.GenerateNewTestClassRecipe$testClassVisitor.visitClassDeclaration(GenerateNewTestClassRecipe.java:101)
	at org.openrewrite.java.tree.J$ClassDeclaration.acceptJava(J.java:1286)
	at org.openrewrite.java.tree.J.accept(J.java:59)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:283)
	at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:366)
	at org.openrewrite.java.JavaVisitor.lambda$visitCompilationUnit$10(JavaVisitor.java:501)
	at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
	at org.openrewrite.java.JavaVisitor.visitCompilationUnit(JavaVisitor.java:501)
	at org.openrewrite.java.JavaIsoVisitor.visitCompilationUnit(JavaIsoVisitor.java:113)
	at org.openrewrite.java.JavaIsoVisitor.visitCompilationUnit(JavaIsoVisitor.java:30)
	at org.openrewrite.java.tree.J$CompilationUnit.acceptJava(J.java:1560)
	at org.openrewrite.java.tree.J.accept(J.java:59)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:283)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:184)
	at org.openrewrite.scheduling.RecipeRunCycle.lambda$editSources$5(RecipeRunCycle.java:164)
	at io.micrometer.core.instrument.AbstractTimer.recordCallable(AbstractTimer.java:175)
	at org.openrewrite.table.RecipeRunStats.recordEdit(RecipeRunStats.java:68)
	at org.openrewrite.scheduling.RecipeRunCycle.lambda$editSources$6(RecipeRunCycle.java:161)
	at org.openrewrite.scheduling.RecipeStack.reduce(RecipeStack.java:57)
	at org.openrewrite.scheduling.RecipeRunCycle.lambda$editSources$7(RecipeRunCycle.java:134)
	at org.openrewrite.internal.InMemoryLargeSourceSet.lambda$edit$0(InMemoryLargeSourceSet.java:66)
	at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
	at org.openrewrite.internal.InMemoryLargeSourceSet.edit(InMemoryLargeSourceSet.java:65)
	at org.openrewrite.scheduling.RecipeRunCycle.editSources(RecipeRunCycle.java:133)
	at org.openrewrite.RecipeScheduler.runRecipeCycles(RecipeScheduler.java:86)
	at org.openrewrite.RecipeScheduler.scheduleRun(RecipeScheduler.java:41)
	at org.openrewrite.Recipe.run(Recipe.java:340)
	at org.openrewrite.Recipe.run(Recipe.java:336)
	at org.openrewrite.Recipe.run(Recipe.java:332)
	at org.openrewrite.maven.AbstractRewriteBaseRunMojo.runRecipe(AbstractRewriteBaseRunMojo.java:142)
	at org.openrewrite.maven.AbstractRewriteBaseRunMojo.listResults(AbstractRewriteBaseRunMojo.java:119)
	at org.openrewrite.maven.AbstractRewriteRunMojo.execute(AbstractRewriteRunMojo.java:63)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:328)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:316)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:174)
	at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:75)
	at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:162)
	at org.apache.maven.buildcache.BuildCacheMojosExecutionStrategy.execute(BuildCacheMojosExecutionStrategy.java:132)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:159)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:906)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:283)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:206)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:283)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:226)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:407)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:348)
Caused by: /home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:107: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:291: org.jdom.input.JDOMParseException: Error on line 1: The markup in the document following the root element must be well-formed.
	at org.apache.tools.ant.ProjectHelper.addLocationToBuildException(ProjectHelper.java:582)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:394)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:197)
	at jdk.internal.reflect.GeneratedMethodAccessor453.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:217)
	at jdk.internal.reflect.GeneratedMethodAccessor453.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:197)
	at jdk.internal.reflect.GeneratedMethodAccessor453.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:391)
	... 72 more
Caused by: /home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:291: org.jdom.input.JDOMParseException: Error on line 1: The markup in the document following the root element must be well-formed.
	at com.sun.ant.taskdefs.xml.XMLFragment.init(XMLFragment.java:134)
	at com.sun.ant.taskdefs.xml.ModifyXML.addConfiguredXmlfragment(ModifyXML.java:220)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.IntrospectionHelper$AddNestedCreator.istore(IntrospectionHelper.java:1519)
	at org.apache.tools.ant.IntrospectionHelper$AddNestedCreator.store(IntrospectionHelper.java:1513)
	at org.apache.tools.ant.IntrospectionHelper$Creator.store(IntrospectionHelper.java:1418)
	at org.apache.tools.ant.UnknownElement.handleChild(UnknownElement.java:601)
	at org.apache.tools.ant.UnknownElement.handleChildren(UnknownElement.java:367)
	at org.apache.tools.ant.UnknownElement.configure(UnknownElement.java:207)
	at org.apache.tools.ant.UnknownElement.maybeConfigure(UnknownElement.java:166)
	at org.apache.tools.ant.Task.perform(Task.java:349)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:197)
	at jdk.internal.reflect.GeneratedMethodAccessor453.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:391)
	... 129 more
Caused by: org.jdom.input.JDOMParseException: Error on line 1: The markup in the document following the root element must be well-formed.
	at org.jdom.input.SAXBuilder.build(SAXBuilder.java:533)
	at org.jdom.input.SAXBuilder.build(SAXBuilder.java:946)
	at com.sun.ant.taskdefs.xml.XMLFragment.init(XMLFragment.java:130)
	... 165 more
Caused by: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 49; The markup in the document following the root element must be well-formed.
	at java.xml/com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:204)
	at java.xml/com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.fatalError(ErrorHandlerWrapper.java:178)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:400)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:327)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLScanner.reportFatalError(XMLScanner.java:1465)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$TrailingMiscDriver.next(XMLDocumentScannerImpl.java:1394)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:112)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:542)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:889)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:825)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1224)
	at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:637)
	at org.jdom.input.SAXBuilder.build(SAXBuilder.java:518)
	... 167 more

INFO: TODO: due to The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:107: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:291: org.jdom.input.JDOMParseException: Error on line 1: The markup in the document following the root element must be well-formed. class ee.jakarta.tck.persistence.core.entitytest.apitests.Client couldn't be processed.
/home/smarlow/tck/tck10/jakartaeetck/src/com/sun/ts/tests/jpa/core/entitytest/apitests/build.xml:44: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:107: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:291: org.jdom.input.JDOMParseException: Error on line 1: The markup in the document following the root element must be well-formed.
	at org.apache.tools.ant.ProjectHelper.addLocationToBuildException(ProjectHelper.java:582)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:394)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.Target.execute(Target.java:449)
	at org.apache.tools.ant.Target.performTasks(Target.java:470)
	at tck.jakarta.platform.ant.PackageTarget.execute(PackageTarget.java:552)
	at tck.jakarta.platform.ant.api.TestPackageInfoBuilder.parseVehiclePackage(TestPackageInfoBuilder.java:308)
	at tck.jakarta.platform.ant.api.TestPackageInfoBuilder.buildTestClientsEx(TestPackageInfoBuilder.java:211)
	at tck.jakarta.platform.ant.api.TestPackageInfoBuilder.buildTestPackgeInfoEx(TestPackageInfoBuilder.java:115)
	at tck.jakarta.platform.rewrite.GenerateNewTestClassRecipe$testClassVisitor.visitClassDeclaration(GenerateNewTestClassRecipe.java:183)
	at tck.jakarta.platform.rewrite.GenerateNewTestClassRecipe$testClassVisitor.visitClassDeclaration(GenerateNewTestClassRecipe.java:101)
	at org.openrewrite.java.tree.J$ClassDeclaration.acceptJava(J.java:1286)
	at org.openrewrite.java.tree.J.accept(J.java:59)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:283)
	at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:366)
	at org.openrewrite.java.JavaVisitor.lambda$visitCompilationUnit$10(JavaVisitor.java:501)
	at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
	at org.openrewrite.java.JavaVisitor.visitCompilationUnit(JavaVisitor.java:501)
	at org.openrewrite.java.JavaIsoVisitor.visitCompilationUnit(JavaIsoVisitor.java:113)
	at org.openrewrite.java.JavaIsoVisitor.visitCompilationUnit(JavaIsoVisitor.java:30)
	at org.openrewrite.java.tree.J$CompilationUnit.acceptJava(J.java:1560)
	at org.openrewrite.java.tree.J.accept(J.java:59)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:283)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:184)
	at org.openrewrite.scheduling.RecipeRunCycle.lambda$editSources$5(RecipeRunCycle.java:164)
	at io.micrometer.core.instrument.AbstractTimer.recordCallable(AbstractTimer.java:175)
	at org.openrewrite.table.RecipeRunStats.recordEdit(RecipeRunStats.java:68)
	at org.openrewrite.scheduling.RecipeRunCycle.lambda$editSources$6(RecipeRunCycle.java:161)
	at org.openrewrite.scheduling.RecipeStack.reduce(RecipeStack.java:57)
	at org.openrewrite.scheduling.RecipeRunCycle.lambda$editSources$7(RecipeRunCycle.java:134)
	at org.openrewrite.internal.InMemoryLargeSourceSet.lambda$edit$0(InMemoryLargeSourceSet.java:66)
	at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
	at org.openrewrite.internal.InMemoryLargeSourceSet.edit(InMemoryLargeSourceSet.java:65)
	at org.openrewrite.scheduling.RecipeRunCycle.editSources(RecipeRunCycle.java:133)
	at org.openrewrite.RecipeScheduler.runRecipeCycles(RecipeScheduler.java:86)
	at org.openrewrite.RecipeScheduler.scheduleRun(RecipeScheduler.java:41)
	at org.openrewrite.Recipe.run(Recipe.java:340)
	at org.openrewrite.Recipe.run(Recipe.java:336)
	at org.openrewrite.Recipe.run(Recipe.java:332)
	at org.openrewrite.maven.AbstractRewriteBaseRunMojo.runRecipe(AbstractRewriteBaseRunMojo.java:142)
	at org.openrewrite.maven.AbstractRewriteBaseRunMojo.listResults(AbstractRewriteBaseRunMojo.java:119)
	at org.openrewrite.maven.AbstractRewriteRunMojo.execute(AbstractRewriteRunMojo.java:63)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:328)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:316)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:174)
	at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:75)
	at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:162)
	at org.apache.maven.buildcache.BuildCacheMojosExecutionStrategy.execute(BuildCacheMojosExecutionStrategy.java:132)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:159)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:906)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:283)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:206)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:283)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:226)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:407)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:348)
Caused by: /home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:107: The following error occurred while executing this line:
/home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:291: org.jdom.input.JDOMParseException: Error on line 1: The markup in the document following the root element must be well-formed.
	at org.apache.tools.ant.ProjectHelper.addLocationToBuildException(ProjectHelper.java:582)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:394)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:197)
	at jdk.internal.reflect.GeneratedMethodAccessor2124.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:217)
	at jdk.internal.reflect.GeneratedMethodAccessor2124.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:197)
	at jdk.internal.reflect.GeneratedMethodAccessor2124.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:391)
	... 72 more
Caused by: /home/smarlow/tck/tck10/jakartaeetck/bin/xml/ts.par.xml:291: org.jdom.input.JDOMParseException: Error on line 1: The markup in the document following the root element must be well-formed.
	at com.sun.ant.taskdefs.xml.XMLFragment.init(XMLFragment.java:134)
	at com.sun.ant.taskdefs.xml.ModifyXML.addConfiguredXmlfragment(ModifyXML.java:220)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.IntrospectionHelper$AddNestedCreator.istore(IntrospectionHelper.java:1519)
	at org.apache.tools.ant.IntrospectionHelper$AddNestedCreator.store(IntrospectionHelper.java:1513)
	at org.apache.tools.ant.IntrospectionHelper$Creator.store(IntrospectionHelper.java:1418)
	at org.apache.tools.ant.UnknownElement.handleChild(UnknownElement.java:601)
	at org.apache.tools.ant.UnknownElement.handleChildren(UnknownElement.java:367)
	at org.apache.tools.ant.UnknownElement.configure(UnknownElement.java:207)
	at org.apache.tools.ant.UnknownElement.maybeConfigure(UnknownElement.java:166)
	at org.apache.tools.ant.Task.perform(Task.java:349)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:197)
	at jdk.internal.reflect.GeneratedMethodAccessor2124.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at java.base/java.util.Vector.forEach(Vector.java:1365)
	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:67)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:391)
	... 129 more
Caused by: org.jdom.input.JDOMParseException: Error on line 1: The markup in the document following the root element must be well-formed.
	at org.jdom.input.SAXBuilder.build(SAXBuilder.java:533)
	at org.jdom.input.SAXBuilder.build(SAXBuilder.java:946)
	at com.sun.ant.taskdefs.xml.XMLFragment.init(XMLFragment.java:130)
	... 165 more
Caused by: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 49; The markup in the document following the root element must be well-formed.
	at java.xml/com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:204)
	at java.xml/com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.fatalError(ErrorHandlerWrapper.java:178)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:400)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:327)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLScanner.reportFatalError(XMLScanner.java:1465)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$TrailingMiscDriver.next(XMLDocumentScannerImpl.java:1394)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:112)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:542)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:889)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:825)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1224)
	at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:637)
	at org.jdom.input.SAXBuilder.build(SAXBuilder.java:518)
	... 167 more

--- a/tools/tck-rewrite-ant/src/main/resources/TsPar.stg
+++ b/tools/tck-rewrite-ant/src/main/resources/TsPar.stg
@@ -16,7 +16,7 @@ genPar(par, testClass) ::= <<
 }#
 #endif#
     // The persistence.xml descriptor
-    URL parURL = #testClass#.class.getResource("/#par.relativeDescriptorPath#");
+    URL parURL = #testClass#.class.getResource("persistence.xml");
     if(parURL != null) {
       #par.typedArchiveName#.addAsManifestResource(parURL, "persistence.xml");
     }

--- a/tools/tck-rewrite-ant/src/test/java/tck/conversion/ant/ParseJpaDescriptors.java
+++ b/tools/tck-rewrite-ant/src/test/java/tck/conversion/ant/ParseJpaDescriptors.java
@@ -1,0 +1,186 @@
+package tck.conversion.ant;
+
+import com.sun.ts.lib.harness.VehicleVerifier;
+import org.apache.tools.ant.MagicNames;
+import org.apache.tools.ant.Project;
+import org.apache.tools.ant.ProjectHelper;
+import org.apache.tools.ant.Target;
+import tck.jakarta.platform.ant.PackageTarget;
+import tck.jakarta.platform.ant.ProjectWrapper;
+import tck.jakarta.platform.ant.api.DefaultEEMapping;
+import tck.jakarta.platform.vehicles.VehicleType;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A one-off CLI app to parse the JPA tests build.xml files from the EE 10 TCK to collect the transformed
+ * persistence.xml files and write them into the EE11 tree with package renaming.
+ */
+public class ParseJpaDescriptors {
+    static String getArg(String name, String defaultValue, String[] args) {
+        String value = defaultValue;
+        for (String arg : args) {
+            if (arg.startsWith(name)) {
+                int equals = arg.indexOf('=');
+                value = arg.substring(equals + 1);
+            }
+        }
+        return value;
+    }
+    public static void main(String[] args) throws Exception {
+        String tsHome = "/home/starksm/Dev/Jakarta/wildflytck/jakartaeetck";
+        String tckRepo = "/home/starksm/Dev/Jakarta/sms-platform-tck";
+        if(args.length != 0) {
+            tsHome = getArg("ts.home", tsHome, args);
+            tckRepo = getArg("tck.repo", tckRepo, args);
+        }
+        if(!Files.exists(Paths.get(tsHome))) {
+            throw new FileNotFoundException(tsHome + " does not exist, pass in a valid EE 10 TCK root directory");
+        }
+        if(!Files.exists(Paths.get(tckRepo))) {
+            throw new FileNotFoundException(tckRepo + " does not exist, pass in a valid EE 11 TCK src repository root");
+        }
+
+        System.setProperty("ts.home", tsHome);
+        ParseJpaDescriptors parser = new ParseJpaDescriptors();
+        parser.run(Path.of(tsHome), Path.of(tckRepo));
+    }
+
+    Path tsHome;
+    Path tckRepo;
+    DefaultEEMapping eeMapping = DefaultEEMapping.getInstance();
+
+    public void run(Path tsHome, Path tckRepo) throws Exception {
+        this.tsHome = tsHome;
+        this.tckRepo = tckRepo;
+        // Create/link directories that are needed to parse the build.xml files
+        System.setProperty("deliverabledir", "tck");
+
+        Path deliveryDir = tsHome.resolve("install/tck");
+        Files.createDirectories(deliveryDir);
+        Path binDir = tsHome.resolve("bin");
+        Path installBinDir = tsHome.resolve("install/tck/bin");
+        if(!Files.isSymbolicLink(installBinDir)) {
+            Files.createSymbolicLink(installBinDir, binDir);
+        }
+        // Scans for all build.xml files under the jpa tree and parses them into ant project instances
+        scanBuildFiles(tsHome.resolve("src/com/sun/ts/tests/jpa"));
+    }
+    /**
+     * Walks the TCK src tree to collect and then parse the ant build.xml files
+     *
+     * @param sourceRoot
+     * @throws IOException
+     */
+    void scanBuildFiles(Path sourceRoot) throws IOException {
+        BuildXmlFilter filter = new BuildXmlFilter();
+        Files.walkFileTree(sourceRoot, filter);
+        List<Path> buildXmls = filter.getBuildFiles();
+        System.out.printf("Found %d build.xml files\n", buildXmls.size());
+        for(Path buildXml : buildXmls) {
+            parseBuildXml(buildXml);
+        }
+    }
+    Project initProject(Path buildXml) {
+        Project project = new Project();
+        project.init();
+        // The location of the glassfish download for the jakarta api jars
+        project.setProperty("ts.home", tsHome.toAbsolutePath().toString());
+        project.setProperty("javaee.home.ri", "${ts.home}/../glassfish7/glassfish");
+        project.setBaseDir(buildXml.getParent().toFile());
+        project.setProperty(MagicNames.ANT_FILE, buildXml.toAbsolutePath().toString());
+        return project;
+    }
+
+    /**
+     * Given a build.xml file path, parse the file using the ant {@link Project}
+     * and {@link ProjectHelper} and print a summary of the parsed {@link PackageTarget}
+     *
+     * @param buildXml
+     */
+    Project parseBuildXml(Path buildXml) throws IOException {
+        Project project = initProject(buildXml);
+        ProjectHelper.configureProject(project, buildXml.toFile());
+        // The package targets are what build the test artifacts
+        Target antPackageTarget = project.getTargets().get("package");
+        if(antPackageTarget == null) {
+            return null;
+        }
+        VehicleVerifier verifier = VehicleVerifier.getInstance(buildXml.toFile());
+        String[] vehicles = verifier.getVehicleSet();
+
+        Path modifiedPersistenceXml = Path.of("/tmp/modified.persistence.xml");
+        // Remove any previously generated persistence.xml
+        Files.deleteIfExists(modifiedPersistenceXml);
+        System.out.printf("Parsing(%s)\n", buildXml);
+        PackageTarget pkgTargetWrapper = new PackageTarget(new ProjectWrapper(project), antPackageTarget);
+        if(vehicles.length == 0) {
+            // Run the ant "package" target
+            pkgTargetWrapper.execute();
+        } else {
+            VehicleType vehicleType = VehicleType.valueOf(vehicles[0]);
+            pkgTargetWrapper.execute(vehicleType);
+        }
+
+        // If a modified persistence.xml was generated, write it to the EE 11 tree
+        if(modifiedPersistenceXml.toFile().exists()) {
+            System.out.printf("Modified persistence.xml: %s\n", modifiedPersistenceXml);
+            mapAndWriteDescriptor(modifiedPersistenceXml, buildXml.getParent());
+        }
+
+        return project;
+    }
+    void mapAndWriteDescriptor(Path persistenceXml, Path testDir) throws IOException {
+        List<String> lines = Files.readAllLines(persistenceXml);
+        StringBuilder tmp = new StringBuilder();
+        for (String line : lines) {
+            // <class>com.sun.ts.tests.jpa.core.annotations.elementcollection.A</class>
+            String mapped = line;
+            if(line.matches("<class>.*</class>")) {
+                int start = line.indexOf("<class>") + "<class>".length();
+                int end = line.indexOf("</class>");
+                String ee10Class = line.substring(start, end);
+                String ee11Class = eeMapping.getEE11Name(ee10Class);
+                mapped = line.replace(ee10Class, ee11Class);
+            }
+            tmp.append(mapped).append("\n");
+        }
+        String testPath = testDir.toString();
+        int idx = testPath.indexOf("com/sun/ts/tests/jpa");
+        String ee10Path = testPath.substring(idx);
+        String ee10Pkg = ee10Path.replace('/', '.');
+        String ee11Pkg = eeMapping.getEE11Name(ee10Pkg);
+        String ee11Path = ee11Pkg.replace('.', '/');
+
+        Path ee11PersistenceXml = null;
+        // jpa/platform-tests/src/main/java
+        Path ee11TestDir1 = tckRepo.resolve("jpa/platform-tests/src/main/java").resolve(ee11Path);
+        if(ee11TestDir1.toFile().exists()) {
+            ee11PersistenceXml = ee11TestDir1.resolve("persistence.xml");
+        } else {
+            // jpa/spec-tests/src/main/java
+            Path ee11TestDir2 = tckRepo.resolve("jpa/spec-tests/src/main/java").resolve(ee11Path);
+            if(!ee11TestDir2.toFile().exists()) {
+                System.err.printf("Neither %s nor %s exist\n", ee11TestDir1, ee11TestDir1);
+                return;
+            }
+            ee11PersistenceXml = ee11TestDir2.resolve("persistence.xml");
+        }
+        // If there is a persistence.xml in the EE 11 tree, write the modified one there with .new extension
+        if(ee11PersistenceXml.toFile().exists()) {
+            Path altPath = ee11PersistenceXml.resolveSibling("persistence.new.xml");
+            Files.writeString(altPath, tmp.toString());
+            System.out.printf("Wrote %s\n", altPath);
+        } else {
+            Files.writeString(ee11PersistenceXml, tmp.toString());
+            System.out.printf("Wrote %s\n", ee11PersistenceXml);
+        }
+
+    }
+}

--- a/tools/tck-rewrite-ant/src/test/java/tck/conversion/ant/ParseJpaDescriptors.java
+++ b/tools/tck-rewrite-ant/src/test/java/tck/conversion/ant/ParseJpaDescriptors.java
@@ -142,7 +142,7 @@ public class ParseJpaDescriptors {
         for (String line : lines) {
             // <class>com.sun.ts.tests.jpa.core.annotations.elementcollection.A</class>
             String mapped = line;
-            if(line.matches("<class>.*</class>")) {
+            if(line.contains("<class>")) {
                 int start = line.indexOf("<class>") + "<class>".length();
                 int end = line.indexOf("</class>");
                 String ee10Class = line.substring(start, end);

--- a/tools/tck-rewrite-ant/src/test/java/tck/conversion/ant/api/DeploymentMethodTest.java
+++ b/tools/tck-rewrite-ant/src/test/java/tck/conversion/ant/api/DeploymentMethodTest.java
@@ -474,4 +474,33 @@ public class DeploymentMethodTest {
         System.out.println(packageInfo.getTestClientFiles());
     }
 
+    @Test
+    public void test_jpa_core_annotations_elementcollection() throws IOException {
+        TestPackageInfoBuilder builder = new TestPackageInfoBuilder(tsHome);
+        List<TestMethodInfo> testMethods = Arrays.asList(
+                new TestMethodInfo("executeTest", "Exception"),
+                new TestMethodInfo("getOutputParameterValueIntIllegalArgumentExceptionTest", "Exception"),
+                new TestMethodInfo("getFirstResultTest", "Exception")
+
+        );
+        Class<?> baseTestClass = ee.jakarta.tck.persistence.core.annotations.elementcollection.Client1.class;
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, DefaultEEMapping.getInstance());
+        System.out.println(packageInfo);
+        System.out.println(packageInfo.getTestClientFiles());
+    }
+    @Test
+    public void test_jpa_core_entitytest_apitests() throws IOException {
+        TestPackageInfoBuilder builder = new TestPackageInfoBuilder(tsHome);
+        List<TestMethodInfo> testMethods = Arrays.asList(
+                new TestMethodInfo("executeTest", "Exception"),
+                new TestMethodInfo("getOutputParameterValueIntIllegalArgumentExceptionTest", "Exception"),
+                new TestMethodInfo("getFirstResultTest", "Exception")
+
+        );
+        Class<?> baseTestClass = ee.jakarta.tck.persistence.core.entitytest.apitests.Client.class;
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, DefaultEEMapping.getInstance());
+        System.out.println(packageInfo);
+        System.out.println(packageInfo.getTestClientFiles());
+    }
+
 }

